app/src/main/java/org/mozilla/fenix/wifi/WifiConnectionMonitor.kt

```diff
@@ -35,16 +35,21 @@ class WifiConnectionMonitor(app: Application) {
 
     private val frameworkListener = object : ConnectivityManager.NetworkCallback() {
         override fun onLost(network: Network?) {
-            callbacks.forEach { it(false) }
+            notifyListeners(false)
             lastKnownStateWasAvailable = false
         }
 
         override fun onAvailable(network: Network?) {
-            callbacks.forEach { it(true) }
+            notifyListeners(true)
             lastKnownStateWasAvailable = true
         }
     }
 
+    internal fun notifyListeners(value: Boolean) {
+        val items = ArrayList(callbacks)
```

Comment:
Can we instantiate this (line 29) as an ArrayList? Or are you doing this here to make it non-mutable?

Reply:
[MutableList](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-mutable-list/) is a more abstract interface, and could be swapped by other implementations, under the hood is backed by a `ArrayList` when we call `mutableListOf` :)

![image](https://user-images.githubusercontent.com/773158/98282032-9a8fa900-1f6b-11eb-967f-e9d2216de7ee.png)

Here we want to create a defensive copy of the initial list, to iterate safely while the original list can be modified.

Reply:
Excellent! Thank you :D